build.psm1

```diff
@@ -447,20 +447,24 @@ function Start-PSBuild {
     }
 
     function Stop-DevPowerShell {
+        if (-not (Test-Path $script:Options.Output)) {
+            return
+        }
+        $pwshDev = Resolve-Path $script:Options.Output
         Get-Process pwsh* |
             Where-Object {
                 $_.Modules |
                 Where-Object {
-                    $_.FileName -eq (Resolve-Path $script:Options.Output).Path
+                    $_.FileName -eq $pwshDev.Path
                 }
             } |
         Stop-Process -Verbose
     }
 
     if ($Clean) {
         Write-Log "Cleaning your working directory. You can also do it with 'git clean -fdx --exclude .vs/PowerShell/v15/Server/sqlite3'"
-        Push-Location $PSScriptRoot
         try {
+            Push-Location $PSScriptRoot
```

Comment:
Since `try` block starts with `Push-Location`, IMO its fine to have it inside.

Reply:
@kvprasoon I think if you try this script, you'll find the results are not as you expect.

```powershell
Write-Output (Get-Location)
Push-Location "~\Documents"
Write-Output (Get-Location)
try
{
    Push-Location 'foo'
}
finally
{
    Write-Output (Get-Location)
    Pop-Location
    Write-Output (Get-Location)
}
```

Reply:
@dantraMSFT 👍

Reply:
@kvprasoon - I realized that Push-Location 'foo' does not throw a terminating exception.  So, if you want this to work reliably, you would need -ErrorAction Stop on the Push-Location call.  Regardless, even if you use -Stop in the try block and it fails, Pop-Location is still attempting to pop a location that never got pushed.

At the end of the day, there's no benefit for the Push-Location changes.

Reply:
@dantraMSFT 
1. I feel that it's more readable having `Push-Location` inside try block, to make it looks like a section of code wrapped by `Push` and `Pop`. You may disagree.

2. The purpose of moving `Push-Location` is to standardise the logic in `build.psm1`, since both style exists in the current script. I can also move `Push-Location` outside try block:
https://github.com/PowerShell/PowerShell/blob/9947985b87dbcffc5cdafe6e412e07ccd4f3f95c/build.psm1#L249-L250
https://github.com/PowerShell/PowerShell/blob/9947985b87dbcffc5cdafe6e412e07ccd4f3f95c/build.psm1#L472-L473

3. Then you might want to realise one more thing:
```PowerShell
Write-Host $(Get-Location -Stack)
Push-Location '~/Documents'
Write-Host $(Get-Location -Stack)
Push-Location 'foo'
Write-Host $(Get-Location -Stack)
```
Now run `Pop-Location`, will you go back to '~'?

Reply:
Currently, Push-Location will always push current directory into the stack, and then try to change to new directory, and that's all. Even if the 'change directory' failed, it will not throw a terminating exception nor pop from the stack to revert the action.
If this is the correct behaviour of Push-Location (e.g. by design), then it should be moved inside the try block with option `-ErrorAction Stop` to stop further action and have Pop-Location in finally block to pop the pushed location. Otherwise it should be moved outside of the try block with option `-ErrorAction Stop` to stop further action which also preventing it to "pop a location that never got pushed".

I do not see how there is no benefit for the Push-Location changes. It just depends on whether the current Push-Location behaviour is correct or not.